Jellyfin.Server.Implementations/FullSystemBackup/BackupService.cs

```diff
@@ -0,0 +1,396 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Database.Implementations;
+using MediaBrowser.Common.Configuration;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.SystemBackupService;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Implementations.Backup;
+
+/// <summary>
+/// Contains methods for creating and restoring backups.
+/// </summary>
+public class BackupService : IBackupService
+{
+    private readonly ILogger<BackupService> _logger;
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+    private readonly IServerApplicationHost _applicationHost;
+    private readonly IServerApplicationPaths _applicationPaths;
+    private readonly IJellyfinDatabaseProvider _jellyfinDatabaseProvider;
+    private static string _manifestEntryName = "manifest.json";
+    private readonly JsonSerializerOptions _serializerSettings = new JsonSerializerOptions()
```

Comment:
```suggestion
        private readonly JsonSerializerOptions _serializerSettings = new(JsonSerializerDefaults.General)
```

Should be the same thing but I like being explicit